server.js

```diff
@@ -45,157 +53,164 @@ function main () {
+                  ])
+                  .then(values => {
+                    var send = mergeInfo(values[0], values[1]);
+                    socket.emit("info sent", send);
+                    called = false;
+                  });
+
+                function mergeInfo(versionChampions, playerInfo){
+                  let toSend = {};
+
+                  let championsMap = versionChampions[1];
+                  let champMasteries = playerInfo[1];
+                  toSend.infos = playerInfo[0];
+
+                  for (let i = 0; i < champMasteries.length; ++i) {
+                    let champId = champMasteries[i].championId;
+                    let urlSquare = "http://ddragon.leagueoflegends.com/cdn/" + versionChampions[0] + "/img/champion/" + championsMap[champId][0] + ".png";
+                    champMasteries[i].name = championsMap[champId][1];
+                    champMasteries[i].riotName = championsMap[champId][0];
+                    champMasteries[i].urlImage = urlSquare;
+
+                    champId = championsMap[champId];
+                  }
+                  toSend.champMasteries = champMasteries;
+                  return toSend;
+                }
+              }
             }
         });
     });
 }
 
-function getGameInfo(ApiKey, toSend, championsMap, region, summonerName, version, callback) {
-    async.waterfall([
-        async.apply(getApiKey, pathToYourApiKey, ApiKey, toSend, championsMap, region, summonerName, version),
-        getVersion,
-        getChampionsJson,
-        getPlayerInfo
-    ], function (err, ApiKey, toSend, championsMap, region, summonerName, version) {
-        callback(null, ApiKey, toSend, championsMap, region, summonerName, version);
-    });
-}
+function getGameInfo(region) {
+  function logError(e) {
+    console.error("error is:" + e);
+    throw (e);
+  }
+  var a = getVersion(region);
+  var b = a.then(function (version) {
+    let req = "http://ddragon.leagueoflegends.com/cdn/" + version + "/data/en_US/champion.json " ;
 
-function getApiKey(file, ApiKey, toSend, championsMap, region, summonerName, version, callback) {
-    ApiKey = require(file);
-    callback(null, ApiKey, toSend, championsMap, region, summonerName, version);
-}
+    return new Promise(function(resolve, reject) {
+      request(req, function (error, response, body) {
+        let championsMap = {};
 
-function getVersion (ApiKey, toSend, championsMap, region, summonerName, version, callback) {
-    let req = "https://global.api.pvp.net/api/lol/static-data/" + region + "/v1.2/versions?api_key=" + ApiKey;
-    request(req, function (error, response, body) {
         if (!error && response.statusCode == 200) {
-            let str = JSON.parse(body);
-            version = str[0];
-            callback(null, ApiKey, toSend, championsMap, region, summonerName, version);
+          let str = JSON.parse(body);
+          delete str.type;
+          delete str.format;
+          delete str.version;
```

Comment:
You don't need the `delete`s statements here as you are redefining `str` as `str.data` and so losing these three properties. You could also just use something like `let str = JSON.parse(body).data`.